docs/designs/topology.md

```diff
@@ -186,128 +240,170 @@ spec:
-  - Operator looks for ClusterTopology with configured name (defaults to "grove-topology")
-  - If ClusterTopology with that name doesn't exist → operator exits with ENOENT and crashloops
-- If `clusterTopology.enabled: false`: topology features disabled
-- Admin must create ClusterTopology with matching name OR disable topology in operator config
+  - Operator generates ClusterTopology CR named "grove-topology"
+  - Operator validates topology config at startup
+  - If config invalid (duplicate domains, invalid keys, etc.) → operator exits with error
+  - If ClusterTopology CR creation fails → operator exits with error
+  - Operator generates KAI Topology CR
+  - If KAI Topology creation fails → reflected in ClusterTopology Ready condition and LastErrors (not operator failure)
+- If `clusterTopology.enabled: false`: topology features disabled, ClusterTopology CR remains (operator does not modify or delete it)
+
+**Configuration Validation:**
+
+At operator startup, Grove validates topology configuration in OperatorConfiguration:
+
+- All domain values must be from predefined set (region, zone, datacenter, block, rack, host, numa)
+- Each domain must be unique within levels list
+- Each key must be unique within levels list
+- Keys must be valid Kubernetes label keys
+- If validation fails → operator exits with descriptive error
+- Note: This validates OperatorConfiguration topology config, not ClusterTopology CR
+- Note: ClusterTopology CR has separate validation layers (API server + webhook - see Validation section above)
 
 **Admin Responsibilities:**
 
-- Manually create Kueue Topology with same name as Grove ClusterTopology for KAI scheduler
-- Ensure topology levels align between Grove ClusterTopology and Kueue Topology
+- Define topology levels in OperatorConfiguration
+- Restart operator when changing topology configuration
+- Grove operator automatically creates and manages both ClusterTopology and KAI Topology CRs
 
 #### Enable/Disable Behavior
 
 **Enabling Topology (clusterTopology.enabled: false → true):**
 
-1. Admin creates ClusterTopology CR with desired topology hierarchy
-2. Admin updates operator config: `clusterTopology.enabled: true`
-3. Admin restarts operator (see Startup Behavior section for details)
-4. Operator validates ClusterTopology CR exists
-5. For existing workloads:
-    - Workloads with topology constraints: operator uses topology from PodCliqueSet label (`grove.io/cluster-topology-name`)
-    - Workloads without topology constraints: no impact (checked via label presence)
-6. For new workloads:
-    - Topology constraints validated against ClusterTopology CR
-   - Mutation webhook adds `grove.io/cluster-topology-name` label to PodCliqueSet (see Mutation Webhook section)
+***For existing workloads:***
+* operator validates constraints, removes constraints referencing non-existent topology levels from PodGang, updates PodCliqueSet status to indicate constraints are ignored
```

Comment:
As per our discussion around this during the requirements phase, can we more explicitly mention putting warnings in the logs as well.  Some warning events on the the effected resources seem like a good idea also given that describing resources and reading through events is a common workflow for SREs.